docs/devel/test-nuclio-locally.md

```diff
@@ -0,0 +1,83 @@
+# Running Nuclio on your computer with local changes
+
+This guide will guide you through the process of setting up Nuclio locally for testing your changes and provide faster development cycles.
+
+#### In This Document
+
+- [Prerequisites](#prerequisites)
+
+- [Running Nuclio](#running-nuclio)
+
+<a id="prerequisites"></a>
+## Prerequisites
+
+Ensure that your setup includes the following prerequisite components:
+
+- Linux or OSX
+- Git
+- Docker version 19.03+
+- The Go toolchain (CI tests with 1.17, best use that)
+- Kubernetes version 1.20+ (mostly for testing) - `minikube` recommended; (you can follow
+  the [Minikube getting-started guide](/docs/setup/minikube/getting-started-minikube.md))
+- Node version 10.x
+- Goland IDE
+
+<a id="running-nuclio"></a>
+## Running Nuclio
+
+1. Install Nuclio CRDs. You can install them by running `test/k8s/ci_assets/install_nuclio_crds.sh`.
+
+
+2. Create a configs folder outside of the Nuclio repo. Inside it, create `configs/platform_config.yaml` with the following body:
```

Comment:
we can add an ignored directory under `hack`, lets say `hack/env` and ignore that folder contents of `hack/env/*.yaml` and add a `hack/env/platform_config.yaml.example` with this content.